nrrd/reader.py

```diff
@@ -339,22 +339,24 @@ def read_data(header, fh=None, filename=None):
     # Get the total number of data points by multiplying the size of each dimension together
     total_data_points = header['sizes'].prod()
 
-    # If encoding is raw and byte skip is -1, then seek backwards to the data
-    # Otherwise skip the number of lines requested
-    if header['encoding'] == 'raw' and byte_skip == -1:
-        fh.seek(-dtype.itemsize * total_data_points, 2)
-    else:
-        for _ in range(line_skip):
-            fh.readline()
-
-    # If a compression encoding is used, then byte skip AFTER decompressing
-    if header['encoding'] == 'raw':
-        # Skip the requested number of bytes and then parse the data using NumPy
+    # Skip the number of lines requested
+    for _ in range(line_skip):
+        fh.readline()    
+            
+    # Skip the requested number of bytes or seek backward, and then parse the data using NumPy
+    if byte_skip < -1:
+        raise Exception('Invalid byteskip, allowed values are greater than or equal to -1')
```

Comment:
Switch to NRRDError rather than Exception
```suggestion
        raise NRRDError('Invalid byteskip, allowed values are greater than or equal to -1')
```

Reply:
I was thinking about that, but than I thought shouldn't it be just Exception? But yeah, we can do that.

Reply:
I'm open to hearing your reasoning, but my thought process is to use NRRDError for any errors in pynrrd. I'm treating NRRDError as the exceptions for the module.